packages/amplify-codegen/src/commands/add.js

```diff
@@ -38,9 +38,9 @@ async function add(context, apiId = null, region = 'us-east-1') {
   }
 
   const schemaPath = ['schema.graphql', 'schema.json'].map(p => path.join(process.cwd(), p)).find(p => fs.existsSync(p));
-  if (withoutInit && !schemaPath) {
+  if (withoutInit && !(apiId || schemaPath)) {
```

Comment:
In the case that `schema.json` is already present when running `amplify codegen add`, the `schema.json` will be overwritten with no warning.